src/app/organization/tellers/allocate-cash/allocate-cash.component.ts

```diff
@@ -0,0 +1,88 @@
+/** Angular Imports. */
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { ActivatedRoute, Router } from '@angular/router';
+import { DatePipe } from '@angular/common';
+
+/** Custom Services. */
+import { OrganizationService } from 'app/organization/organization.service';
+
+/**
+ * Allocate Cash component.
+ */
+@Component({
+  selector: 'mifosx-allocate-cash',
+  templateUrl: './allocate-cash.component.html',
+  styleUrls: ['./allocate-cash.component.scss']
+})
+export class AllocateCashComponent implements OnInit {
+
+  /** Cashier data. */
+  cashierData: any;
+  /** Cashier Form. */
+  allocateCashForm: FormGroup;
+  /** Minimum Date allowed. */
+  minDate = new Date(2000, 0, 1);
+  /** Maximum Date allowed. */
+  maxDate = new Date();
+
+  /**
+   * Get cashier data from `Resolver`.
+   * @param {FormBuilder} formBuilder Form Builder.
+   * @param {ActivatedRoute} route ActivateRoute.
+   * @param {DatePipe} datePipe Date Pipe.
+   * @param {OrganizationService} organizationService Organization Service.
+   * @param {Router} router Router.
+   */
+  constructor(private formBuilder: FormBuilder,
+              private route: ActivatedRoute,
+              private datePipe: DatePipe,
+              private organizationService: OrganizationService,
+              private router: Router) {
+    this.route.data.subscribe((data: { cashierTemplate: any}) => {
+      this.cashierData = data.cashierTemplate;
+    });
+  }
+
+  ngOnInit() {
+    this.setCashierForm();
+  }
+
+  /**
+   * Set Cashier form.
+   */
+  setCashierForm() {
+    this.allocateCashForm = this.formBuilder.group({
+      'office': [{value: this.cashierData.officeName, disabled: true}],
+      'tellerName': [{value: this.cashierData.tellerName, disabled: true}],
+      'cashier': [{value: this.cashierData.cashierName, disabled: true}],
+      'assignmentPeriod': [{value: this.cashierData.startDate && new Date(this.cashierData.startDate), disabled: true}],
```

Comment:
Assignment period is from start date to end date, `start date - end date`

Reply:
Isn't it the start date?

Reply:
no see ui in commy app